### DISCUSSION TITLE
Speed of Compiled Code

### DISCUSSION THREAD

**Discussion by dmrvb** [twinbasic/twinbasic#139] [General]
Not really a discussion, just a bit of interesting info. (because I am very excited about TwinBasic).

My interest is signal processing so have just got some FFT  (Fast Fourier Transform) code running in TwinBasic to compare with fully compiled VB6 code. 
Execution time for 100 16k FFTs: ):......
VB6 (standard) 1.173 seconds
VB6 (optimised) 0.313 seconds
TwinBasic  2.19 seconds.

VB6 compile option for fast code selected. The optimised timing is with array bounds checking and numeric overflow checking etc. disabled. The array bounds is the big gain. I don't know what TwinBasic is doing but my guess is that it is doing bounds checking.

EXE sizes about 64k for the VB6  164k for TwinBasic.

**Comment by WaynePhillipsEA** [comment]
Thanks for posting this.  That seems quite reasonable at the moment since we're only compiling to completely unoptimised debug-grade native code.  At the moment tB always does bounds checking and overflow checking etc, but we will be adding options to turn them off soon.

Wait until we add LLVM support for full on optimized compilations and we'll leave VB6 for dust 😉

**Comment by WaynePhillipsEA** [comment]
It would be great to see the timings when you compile the VB6 EXE to p-code, which would also be roughly the same as running the code in the IDE or even from VBA

**Reply by dmrvb** [reply]
With Pcode compilation and in the IDE the timings are more variable, which is interesting.

based on just 3 or 4 runs....
IDE  between 4 and 4.9 seconds
Pcode  between 4 and 5.6 seconds

**Reply by WaynePhillipsEA** [reply]
Nice, so tB debug compilation sits between VB6 native code and p-code, which is pretty much what I would expect at this stage.  Thanks for confirming!
